### DISCUSSION TITLE
How to change base url

### DISCUSSION THREAD

**Discussion by LucianMihalache** [nuxt-community/auth-module#1581] [Q&A] [answered]
Hello,
I have multiple api's and have the axios's baseUrl set to an api
But I have the authentication api on a different domain
I have tried passing baseUrl config when calling loginWith
```
this.nuxt.$auth.loginWith('apiLogin', {
      data: {
        username: username,
        password: password,
      },
      baseUrl: this.auth_api_url,
      withCredentials: true
    })
```
But it does not seem that its working.
The apiLogin schema extends the LocalSchema and I overwrite just the updateTokens method (because I have multiple tokens)

So, my question is, how can I change the baseUrl for this axios instance, without changing the main axios configuration.

I have tried ovewriting the Auth class in the constructor, but it does not seem to work either

```
export default class ApiLogin extends LocalScheme {

  constructor($auth, options, ...defaults) {
    $auth.ctx.$axios = $auth.ctx.$axios.create({
      withCredentials: true,
      baseURL: $auth.options.baseURL
    })
    super($auth, options, ...defaults);
  }
}
```
$auth.options.baseURL is set up in nuxt.config.js
```
  auth: {
     baseURL: process.env.AUTH_API_URL,
  }
```

**Comment by Intevel** [comment]
Currently there is no way to do this, the Auth module accesses the Axios Instance and does not use its own, the only way is to change the Axios configuration, nothing else comes to my mind.

**Comment by LucianMihalache** [comment, marked as answer]
Hello
I found a solution to this:
You can specify what extra options can be passed to Auth's Axios directly from the nuxt-config.js
```
  auth: {
    plugins: [ '~/plugins/add-login-token.js' ],
    strategies: {
      apiLogin: {
        scheme: '~/schemes/apiLogin',
        endpoints: {
          login: {
            baseURL: process.env.API_URL,
            withCredentials: true,
            url: '/login',
            method: 'post'
          },
          logout: {
            baseURL: process.env.API_URL,
            withCredentials: true,
            url: '/logout',
            method: 'post'
          },
          user: {
            baseURL: process.env.API_URL,
            withCredentials: true,
            url: '/user',
            method: 'get'
          }
        },
        user: {
          property: 'data',
          autoFetch: true
        },
      },
    },
```

**Comment by hvitis** [comment]
You need to change **[@nuxt/axios](https://axios.nuxtjs.org/options/#baseurl)** settings in **nuxt.config.js** file:

LH_BACKEND_URL = 'localhost:8000'
BASE_URL = 'https://your-prod-api-url.com'
API_URL = 'https://your-prod-api-url.com'

```
axios: {
    baseURL: process.env.BASE_URL || LH_BACKEND_URL,
    https: true,
    credentials: true,
    proxy: true,
    prefix: process.env.NODE_ENV === 'production' ? process.env.API_URL : LH_BACKEND_URL
  },
  proxy: {
    '/api': {
      target: process.env.BASE_URL,
      pathRewrite: {
        '^/api/': '/',
      },
    },
  },
  ```
